### PR TITLE
Nikhil_Assignment_21_Done_(3+10=)13Min

### PR DESCRIPTION
**Pull request by NikhilMJun20**
->Completed and pushed Assignment # 21
->Please review and comment.

->PFB the screenshots of the output of the programs:-

**1**
Annotation 2020-06-30 155255
Annotation 2020-06-30 155212

**2**
Output_FibonacciN

### CODE REVIEW

File: src/nikhilMJun20/FibonacciN.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+package nikhilMJun20;
+
+import java.util.Scanner;
+
+public class FibonacciN {
+	void displayFibonacci(int order) {
+		System.out.println("\nFibonacci Series of Order "+order+":-\n");
+		int num1=1,num2=1,num3=0;
+		for(int index=0; index<=order; index++) {
+			if(index<2)
```
Comment (VaishnaviVJun20) [inline src/nikhilMJun20/FibonacciN.java:25]: --You can assign 0 to num1 variable and then there is no need for this if else logic so you can reduce some lines of code.

File: src/nikhilMJun20/NumberSwap.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+		System.out.println("Originally Given Numbers:-\nNumber1 -> "+num1+"     Number2 -> "+num2);
+		num1 += num2;
+		num2 = num1 - num2;
+		num1 -= num2;
+		System.out.println("\nSwapped Numbers:-\nNumber1 -> "+num1+"     Number2 -> "+num2);
+	}
+	
+	public static void main(String[] args) {
+		int num1=25, num2=35;
+		new NumberSwap().swapNum(num1, num2);
```
Comment (VaishnaviVJun20) [inline src/nikhilMJun20/NumberSwap.java:19]: --Code looks good to me, I like how you directly called method.

**Review by VaishnaviVJun20 — COMMENTED** [review]
--Please work on comments.

File: src/nikhilMJun20/NumberSwap.java (hunk 2 above)
Comment (NikhilMJun20) [reply src/nikhilMJun20/NumberSwap.java:19]: Thank you, Vaishnavi!

File: src/nikhilMJun20/FibonacciN.java (hunk 1 above)
Comment (NikhilMJun20) [reply src/nikhilMJun20/FibonacciN.java:25]: ->Sure. I can make that change which is more efficient.
But this requires additional print statement outside the loop - which is kind of like an adjustment made by us without implementing logic of Fibonacci series. Logic should be like unless there are 2 elements which we can add, we should not proceed to 3rd element. 
Let me know your thoughts on this.

Comment (VaishnaviVJun20) [reply src/nikhilMJun20/FibonacciN.java:25]: -- Well if logically thinking this makes sense.
--However if we implement suggested logic, code can be little efficient.
-- I am also not sure which approach is better, so let's our reviewer take the call.

**Review by VaishnaviVJun20 — APPROVED** [review]
--Code looks good to me.
--Keep it up

File: src/nikhilMJun20/FibonacciN.java (hunk 1 above)
Comment (KrishnaKTechnocredits) [reply src/nikhilMJun20/FibonacciN.java:25]: void printFibonacciNumber(int number) {
		int num =0; 
		int num1 = 0, num2=1;
		for(int index=1; index<=5; index++) {
			System.out.print(num2+" ");
			num = num1 + num2;
			num1= num2;
		    num2 = num;
		}
	}
Nikhil if block is not required.

**Pull request merged**